x/gamm/keeper/pool.go

```diff
@@ -298,6 +298,16 @@ func (k Keeper) GetTotalPoolLiquidity(ctx sdk.Context, poolId uint64) (sdk.Coins
 	return pool.GetTotalPoolLiquidity(ctx), nil
 }
 
+// GetTotalPoolShares returns the total number of pool shares for the given pool.
+func (k Keeper) GetTotalPoolShares(ctx sdk.Context, poolId uint64) (sdk.Int, error) {
```

Comment:
Note to reviewer: implementing this helper was necessary to avoid import cycles that came up when interacting with lower levels of GAMM abstractions